### PR TITLE
Add a tiny service worker for teh offlinez

### PR DESCRIPTION
**Pull request by surma**
* Moved `sw.js` to root because scope
* Cached `core.css` for now (can be removed later obv)
* Did *not* cache the webfont but added sensible fallbacks instead

### CODE REVIEW

**Comment by surma** [comment]
Don’t merge yet. Didn’t test permalinks and they don’t work.

File: src/scripts/sw.js [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+    var req = event.request;
+    return event.respondWith(
+      caches
+        .match(req)
+        .then(function(response) {
+          if(response) {
+            return response;
+          }
+
+          return caches.match('/index.html');
```
Comment (paullewis) [inline src/scripts/sw.js]: Wonder if we should use the list of properties here, so we don't let /something-that-does-not-exist be requested. I need to do the same on the server-side.

File: src/styles/core.scss [hunk 2]
```diff
@@ -23,7 +23,7 @@ html, body {
   height: 100%;
   width: 100%;
   min-width: 320px;
-  font-family: 'Roboto';
+  font-family: 'Roboto', Helvetica, sans-serif;
```
Comment (paullewis) [inline src/styles/core.scss]: This should probs be a separate PR. It's good.

File: src/scripts/sw.js (hunk 1 above)
Comment (surma) [inline src/scripts/sw.js]: Agreed.

File: src/styles/core.scss (hunk 2 above)
Comment (surma) [inline src/styles/core.scss]: Okay.

File: src/scripts/css-triggers-core.js [hunk 3]
```diff
@@ -18,3 +18,5 @@
 import AppController from './controller/AppController';
 
 new AppController();
+
+navigator.serviceWorker.register('/sw.js', {scope: '/'});
```
Comment (paullewis) [inline src/scripts/css-triggers-core.js]: Wrap this in a check for SW:

```
if (!('serviceWorker' in navigator))
  return;
```
Though you should probably then wrap this up into a ServiceWorkerController first.

File: src/scripts/sw.js (hunk 1 above)
Comment (surma) [inline src/scripts/sw.js]: Although, on second thought, do we really want this app to 404 ever? I feel like just opening the non-expanded list on invalid property names is acceptable behavior. WDYT?

File: src/scripts/css-triggers-core.js (hunk 3 above)
Comment (paullewis) [inline src/scripts/css-triggers-core.js]: i.e. 
```javascript
import ServiceWorkerController from './controller/ServiceWorkerController';
new ServiceWorkerController();
```

then in the constructor just do an early bail if we don't have a browser that supports SW.

File: src/scripts/sw.js [hunk 4]
```diff
@@ -0,0 +1,51 @@
+(function() {
```
Comment (paullewis) [inline src/scripts/sw.js]: Add the license plz.

File: src/scripts/sw.js (hunk 1 above)
Comment (paullewis) [inline src/scripts/sw.js]: Yes, we should 404 for non-valid URLs. I can file an issue for us to have a nice 404.html that we can cache and serve from the SW.

**Comment by surma** [comment]
@paullewis All done :)

**Comment by paullewis** [comment]
LGTM!

**Comment by surma** [comment]
:tada:
